src/libexpr/primops.cc

```diff
@@ -1836,6 +1836,39 @@ static RegisterPrimOp primop_readDir({
     .fun = prim_readDir,
 });
 
+/* Extend single element string context with another output. */
+static void prim_outputOf(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    SingleDerivedPath drvPath = state.coerceToSingleDerivedPath(pos, *args[0], "while evaluating the first argument to builtins.outputOf");
+
+    std::string_view outputName = state.forceStringNoCtx(*args[1], pos, "while evaluating teh second argument to builtins.outputOf");
+
+    NixStringContextElem::Built path2 {
+        .drvPath = make_ref<SingleDerivedPath>(drvPath),
+        .output = std::string { outputName },
+    };
+
+    v.mkString(
+        DownstreamPlaceholder::fromSingleDerivedPathBuilt(path2).render(),
+        { path2 });
+}
+
+static RegisterPrimOp primop_outputOf({
+    .name = "__outputOf",
+    .args = {"drv path", "output name"},
+    .doc = R"(
+      Return path (actually placeholder path) to the output of the given drv.
```

Comment:
```suggestion
      Return the output path of a derivation, in the form of a placeholder.
```

Also needs a link to what a placeholder is.